test/Interop/Cxx/operators/non-member-out-of-line.swift

```diff
@@ -0,0 +1,50 @@
+// RUN: %empty-directory(%t)
+// RUN: %target-clang -c %S/Inputs/non-member-out-of-line.cpp -I %S/Inputs -o %t/non-member-out-of-line.o
+// RUN: %target-build-swift %s -I %S/Inputs -o %t/non-member-out-of-line %t/non-member-out-of-line.o -Xfrontend -enable-cxx-interop
+// RUN: %target-codesign %t/non-member-out-of-line
+// RUN: %target-run %t/non-member-out-of-line
+//
+// REQUIRES: executable_test
+
+import NonMemberOutOfLine
+import StdlibUnittest
+
+var OperatorsTestSuite = TestSuite("Operators")
+
+OperatorsTestSuite.test("plus") {
+  let lhs = IntBox(value: 42)
+  let rhs = IntBox(value: 23)
+
+  let result = lhs + rhs
+
+  expectEqual(65, result.value)
+}
+
+OperatorsTestSuite.test("minus") {
+  let lhs = IntBox(value: 42)
+  let rhs = IntBox(value: 23)
+
+  let result = lhs - rhs
+
+  expectEqual(19, result.value)
+}
+
+OperatorsTestSuite.test("star") {
+  let lhs = IntBox(value: 42)
+  let rhs = IntBox(value: 23)
+
+  let result = lhs * rhs
+
+  expectEqual(966, result.value)
+}
+
+OperatorsTestSuite.test("slash") {
+  let lhs = IntBox(value: 42)
+  let rhs = IntBox(value: 23)
+
+  let result = lhs / rhs
+
+  expectEqual(1, result.value)
+}
```

Comment:
The tests seem a bit redundant -- do we really need all four of these operators?

Why would a single operator not be enough -- in other words, what's the difference in test coverage between the tests?

Reply:
Well, yes, I deliberated that a bit as well. Happy to drop if you prefer.

The background is that there are quite a few different [operators](https://github.com/apple/llvm-project/blob/swift/master/clang/include/clang/Basic/OperatorKinds.def#L58). Some of them are unary, some binary. Some are allowed as member functions, some are not. Some we will import in different ways (`operator[]` as subscript), some not at all.

So, there will be code that treats different operators differently. Of course not these four. But there will be different code branches. And I want to be sure that every imported operator gets treated right. I guess that the inline version of the tests may sufficient. I added this as a bit extra precaution.

Wdyt?

Reply:
That makes sense in principle, but I believe the four operators you've got here are all treated the same from the point of import -- right?

I would suggest doing one of these:
*    Include just `operator+` and rely on the inline version of the test for coverage of operators that import differently.
*    Include multiple operators, but choose ones that import differently. If this takes a lot of code, it's probably not worth doing though.